packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -202,7 +202,13 @@ abstract class PrimaryPointerGestureRecognizer extends OneSequenceGestureRecogni
         resolve(GestureDisposition.rejected);
         stopTrackingPointer(primaryPointer);
       } else {
-        handlePrimaryPointer(event);
+        try {
+          handlePrimaryPointer(event);
+        } catch (e) {
+          resolve(GestureDisposition.rejected);
+          stopTrackingPointer(primaryPointer);
+          rethrow;
```

Comment:
Oh, i see.  You only want to do this on an error...  That seems like an odd API.